pkg/virtctl/credentials/addkey/addkey.go

```diff
@@ -135,21 +135,32 @@ func addSecretWithSSHKey(
 	}
 
 	accessCredential := newAccessCredential(secret.Name, cmdFlags.User)
-	accessCredentialPatch := patchToAddAccessCredential(accessCredential)
-
+	accessCredentialPatch, err := patch.New(patch.WithAdd("/spec/template/spec/accessCredentials/-", accessCredential)).GeneratePayload()
+	if err != nil {
+		return err
+	}
 	// First, Try to add the new access credential to the existing array.
 	_, err = cli.VirtualMachine(vm.Namespace).Patch(cmd.Context(),
 		vm.Name,
 		types.JSONPatchType,
-		common.MustMarshalPatch(accessCredentialPatch),
+		accessCredentialPatch,
 		metav1.PatchOptions{})
+	if err == nil {
+		return nil
+	}
+
+	// If it fails, it probably means that the array is nil. Try to add the array.
+	fullPatch, err := patch.New(
+		patch.WithTest("/spec/template/spec/accessCredentials", nil),
```

Comment:
Makes sense to have a const for this?

Reply:
I thought about it but there are 2 strings, one for the `accessCredentials` and one for adding the element in the array

Reply:
But I can add one for the path

Reply:
In the end, I added 2 constants.